broker/fragment/store_azure.go

```diff
@@ -29,30 +35,48 @@ type azureBackend struct {
 	endpoint    string
 	accountName string
 	client      pipeline.Pipeline
-	credentials *azblob.SharedKeyCredential
+	svcClient   service.Client
 	clientMu    sync.Mutex
 }
 
 func (a *azureBackend) Provider() string {
 	return "azure"
 }
 
+// https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/service/examples_test.go#L285
 func (a *azureBackend) SignGet(ep *url.URL, fragment pb.Fragment, d time.Duration) (string, error) {
 	cfg, _, err := a.azureClient(ep)
 	if err != nil {
 		return "", err
 	}
 	blobName := cfg.rewritePath(cfg.prefix, fragment.ContentPath())
-	sasQueryParams, err := azblob.BlobSASSignatureValues{
-		Protocol:      azblob.SASProtocolHTTPS, // Users MUST use HTTPS (not HTTP)
-		ExpiryTime:    time.Now().UTC().Add(d), // Timestamps are expected in UTC https://docs.microsoft.com/en-us/rest/api/storageservices/create-service-sas#service-sas-example
+
+	// Set current and past time and create key
+	currentTime := time.Now().UTC().Add(-10 * time.Second)
+	expTime := currentTime.Add(d)
+	info := service.KeyInfo{
+		Start:  to.Ptr(currentTime.UTC().Format(sas.TimeFormat)),
+		Expiry: to.Ptr(expTime.UTC().Format(sas.TimeFormat)),
+	}
+
+	udc, err := a.svcClient.GetUserDelegationCredential(context.Background(), info, nil)
```

Comment:
This is a network request, right? If so, we'd need to root this out and reuse, as we really don't want that on the path here. This method should be _fast_.

It seems like the start / expiry of the delegation key, vs the start / expiry of the individual signatures, are being conflated. We can get a much longer lived delegation key and reuse it over and over.

Reply:
Yuuup, good point. Updated to fetch this when it's expiring soon, otherwise return the cached value